mmcls/models/heads/multi_task_head.py

```diff
@@ -9,35 +9,46 @@
 from .base_head import BaseHead
 
 
+def mmtask_convertor(func, task_name, data_samples):
+    target_type = func.__annotations__['data_samples'].__args__[0].__name__
+    data_samples = data_samples.to_target_data_sample(target_type, task_name)
+
+    return data_samples
+
+
 @MODELS.register_module()
 class MultiTaskHead(BaseHead):
     """Multi task head.
 
     Args:
-        sub_heads (dict): Sub heads to use, the key will be use to rename the
+        task_heads (dict): Sub heads to use, the key will be use to rename the
             loss components.
         common_cfg (dict): The common settings for all heads. Defaults to an
             empty dict.
         init_cfg (dict, optional): The extra initialization settings.
             Defaults to None.
     """
 
-    def __init__(self, sub_heads, common_cfg=dict(), init_cfg=None):
+    def __init__(self, task_heads, common_cfg=dict(), init_cfg=None):
         super(MultiTaskHead, self).__init__(init_cfg=init_cfg)
 
-        assert isinstance(sub_heads, dict), 'The `sub_heads` argument' \
+        assert isinstance(task_heads, dict), 'The `task_heads` argument' \
             "should be a dict, which's keys are task names and values are" \
             'configs of head for the task.'
 
-        self.sub_heads = ModuleDict()
+        self.task_heads = ModuleDict()
 
-        for task_name, head_cfg in sub_heads.items():
-            sub_head = MODELS.build(head_cfg, default_args=common_cfg)
-            self.sub_heads[task_name] = sub_head
+        for task_name, head_cfg in task_heads.items():
+            if head_cfg['type'] != 'MultiTaskHead':
+                sub_head = MODELS.build(head_cfg, default_args=common_cfg)
+            else:
+                sub_head = MODELS.build(
+                    head_cfg, default_args={'common_cfg': common_cfg})
+            self.task_heads[task_name] = sub_head
```

Comment:
Please apply the mmtask_convertor directly to the head here as suggested on slack by haochen ye on openmmlab slack